### PR TITLE
CORE-15310: Update slf4j to v2

### PR DESCRIPTION
**Pull request by josephzunigadaly**
Linked PR: https://github.com/corda/corda-runtime-os/pull/6150

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 kotlinVersion = "1.8.21"
 
 # Logging
-slf4jVersion = { strictly = "1.7.36" }
+slf4jVersion = "2.0.12"
```
Comment (vkolomeyko) [inline gradle/libs.versions.toml]: We probably need to bump the version of `corda-api` on this instance, as this is represents a substantial change, which is arguably of the breaking nature.

**Review by vkolomeyko — APPROVED** [review]
LGTM

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 kotlinVersion = "1.8.21"
 
 # Logging
-slf4jVersion = { strictly = "1.7.36" }
+slf4jVersion = "2.0.13"
```
Comment (driessamyn) [inline gradle/libs.versions.toml:5]: Does this not break compatibility for CorDapps ?

Comment (vkolomeyko) [reply gradle/libs.versions.toml:5]: No, it does not.

**Comment by filipesoliveira** [comment]
Hi @driessamyn,

Are you happy with the current state of the PR? 

Runtime-os, e2e and QA tests are green.

**References:**
QA tests - https://ci05.dev.r3.com/blue/organizations/jenkins/QA%2FCRAFT5%2FCraft_Tests%2FC5%2FLatest%2FOpensource%2FEKS%2FOS-qa-regpack-tests-flow-eks/detail/main/23/pipeline/
e2e tests - https://ci02.dev.r3.com/blue/organizations/jenkins/Corda5%2Fcorda-rebuild-e2e-tests/detail/PR-6150/13/pipeline
runtime-os - https://github.com/corda/corda-runtime-os/pull/6150

**Review by owenstanford — APPROVED** [review]

**Pull request merged**